ctapipe/io/astropy_helpers.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import tables
+from astropy.table import Table
+from astropy.units import Unit
+
+__all__ = ["h5_table_to_astropy"]
+
+
+def h5_table_to_astropy(h5file, path) -> Table:
+    """Get a table from a ctapipe-format HDF5 table as an `astropy.table.Table`
+    object, retaining units. This uses the same unit storage convention as
+    defined by the `HDF5TableWriter`, namely that the units are in attributes
+    named by `<column name>_UNIT` that are parsible by `astropy.units`.
+
+    Parameters
+    ----------
+    h5file: Union[str, tables.file.File]
+       input filename or PyTables file handle
+    path: str
+       path to table in the file
+
+    """
+
+    should_close_file = False
+    if not isinstance(h5file, tables.file.File) and isinstance(h5file, str):
+        h5file = tables.open_file(h5file)
+        should_close_file = True
+    else:
+        raise ValueError(
+            f"expected a string or PyTables filehandle for argument 'h5file', got {h5file}"
+        )
+
+    table: tables.table.Table = h5file.get_node(path)
+
+    column_units = {}  # mapping of colname to unit
+    for attr in table.attrs._f_list():
+        if attr.endswith("_UNIT"):
+            colname = attr[:-5]
+            column_units[colname] = table.attrs[attr]
+
+    astropy_table = Table(table[:])
```

Comment:
You can give the table constructor a `units=` dictionary mapping column names to units

Reply:
doesn't seem to work for QTables

Reply:
Works for me:

```
In [3]: QTable({'x': [1, 2, 3]},  units={'x': 'm'})
Out[3]: 
<QTable length=3>
   x   
   m   
float64
-------
    1.0
    2.0
    3.0
```

Reply:
what version are you using? In my case it fails

Reply:
4.0.1.post1

Reply:
really? Cutting and pasting your example fails for me. 
```
In [6]: astropy.__version__
Out[6]: '4.0.1.post1'
```
```
In [2]: from astropy.table import QTable

In [3]: QTable({'x': [1, 2, 3]},  units={'x': 'm'})
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-43bbf31be07a> in <module>
----> 1 QTable({'x': [1, 2, 3]},  units={'x': 'm'})

~/anaconda3/envs/cta-dev/lib/python3.7/site-packages/astropy/table/table.py in __init__(self, data, masked, names, dtype, meta, copy, rows, copy_indices, **kwargs)
    505         elif kwargs:
    506             raise TypeError('__init__() got unexpected keyword argument {!r}'
--> 507                             .format(list(kwargs.keys())[0]))
    508
    509         if (isinstance(data, np.ndarray) and

TypeError: __init__() got unexpected keyword argument 'units'
```

Reply:
Ah, sorry. Got confused with my envs. I had the dev version of astropy installed into my user site so the output of `conda list astropy` showed the wrong version.